### DISCUSSION TITLE
Custom Model Generator

### DISCUSSION THREAD

**Discussion by shanyssys** [open-webui/open-webui#933] [Ideas]
Lately I've run into models that are having issues loading into Ollama, they usually end up with "out of memory".  For example 7b models like Gemma using 3070ti.

The solution to load 7b models into ollama is to restrict the number of layers which pytorch is loading into the gpu,
this is done by using a flag ```PARAMETER num_gpu 25``` in the Modelfile.

which will look like this when ollama runs the model in log:
```llm_load_tensors: offloading 25 repeating layers to GPU```

As a feature request, I'd like to see a "Set layers for model X" in Open-WebUI which then creates a new model file, uses the ollama create flag and have it all listed. this will help benchmark older cards with less gpu than just rely on the cpu with ```num_gpu 0``` flag.

**Comment by justinh-rahb** [comment]
To clarify, you mean adding the field here in the Modelfile generator?

<img width="756" alt="Screenshot 2024-02-27 at 1 19 08 AM" src="https://github.com/open-webui/open-webui/assets/52832301/17106bb7-0983-4a36-b62b-682236e74166">

**Reply by Gally-Youko** [reply]
I want this too
https://github.com/open-webui/open-webui/issues/1749

**Comment by shanytc** [comment]
Yeah, if possible to add the num_gpu. 
If it's 0 it means cpu, otherwise it will load it the specified amount layers to the gpu

**Reply by gigascake** [reply]
Using this method, Enable Load Model to CPU(DRAM) + VRAM?
